includes/sanitizers/class-amp-core-theme-sanitizer.php

```diff
@@ -570,14 +569,18 @@ public function add_smooth_scrolling( $link_xpaths ) {
 	 * @link https://caniuse.com/#feat=svg
 	 */
 	public function force_svg_support() {
-		$this->dom->documentElement->setAttribute(
-			'class',
-			preg_replace(
-				'/(^|\s)no-svg(\s|$)/',
-				' svg ',
-				$this->dom->documentElement->getAttribute( 'class' )
-			)
-		);
+		$class = $this->dom->documentElement->getAttribute( 'class' );
+
+		if ( strpos( $class, 'no-svg' ) !== false ) {
+			$this->dom->documentElement->setAttribute(
+				'class',
+				preg_replace(
+					'/(^|\s)no-svg(\s|$)/',
+					' svg ',
+					$this->dom->documentElement->getAttribute( 'class' )
+				)
+			);
+		}
```

Comment:
I suggest something like this to optimize/harden this a bit more:

```php
$class = $this->dom->documentElement->getAttribute( 'class' );
if ( $class ) {
	$count = 0;
	$class = preg_replace(
		'/(^|\s)no-svg(\s|$)/',
		' svg ',
		$class,
		$count
	);
	if ( $replaced_count > 0 ) {
		$this->dom->documentElement->setAttribute( 'class', $class );
	}
}
```

Reply:
Done, thanks for the suggestion. You missed the argument for the limit, so I've set it to the default (`-1`).